### PR TITLE
chore(connector-core,core,phrases): use try catch on JSON.parse

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

Use try catch on JSON.parse to avoid unexpected behaviors.

## Testing

UTs.

### CODE REVIEW

**Review by gao-sun — CHANGES_REQUESTED** [review]
wrong base?

**Comment by wangsijie** [comment]
@darcyYe could you provide more info on why wrapping every JSON.parse?

File: packages/connector-core/src/index.ts [hunk 1]
```diff
@@ -11,3 +11,16 @@ export function validateConfig<T>(config: unknown, guard: ZodType): asserts conf
     throw new ConnectorError(ConnectorErrorCodes.InvalidConfig, result.error);
   }
 }
+
+export const jsonSafeParse = (
+  jsonString: string,
+  errorCode: ConnectorErrorCodes = ConnectorErrorCodes.InvalidResponse,
+  payload?: unknown
```
Comment (wangsijie) [inline packages/connector-core/src/index.ts:18]: maybe something like `errorPayload`?

**Review by wangsijie — APPROVED** [review]
Overall LGTM

File: packages/connector-core/src/index.ts (hunk 1 above)
Comment (darcyYe) [reply packages/connector-core/src/index.ts]: updated.

**Review by gao-sun — APPROVED** [review]

**Pull request merged**